app/models/split.rb

```diff
@@ -17,13 +17,50 @@ class Split < ActiveRecord::Base
   validate :registry_weights_must_be_integers
   validate :registry_must_have_winning_variant_if_decided
 
+  enum platform: %i(mobile desktop)
+
   before_validation :cast_registry
 
+  scope :for_presentation, ->(app_build: nil) do
+    app_build.present? ? for_app_build(app_build) : active
+  end
+
+  scope :for_app_build, ->(app_build) do
+    active(as_of: app_build.built_at)
+      .with_feature_incomplete_knockouts_for(app_build)
+  end
+
   scope :active, ->(as_of: nil) do
     as_of ? where('splits.finished_at is null or splits.finished_at > ?', as_of) : where(finished_at: nil)
   end
 
-  enum platform: %i(mobile desktop)
+  scope :with_feature_incomplete_knockouts_for, ->(app_build) do
+    previous_selects = all.arel.projections
+    except(:select)
+      .select(
+        previous_selects,
+        Arel::SelectManager.new
+      .where(arel_excluding_incomplete_features_for(app_build).not)
+          .exists
+          .as('feature_incomplete')
+      )
+      .readonly
+  end
+
+  scope :excluding_incomplete_features_for, ->(app_build) do
+    where(arel_excluding_incomplete_features_for(app_build))
+  end
+
+  class << self
+    private
+
+    def arel_excluding_incomplete_features_for(app_build)
```

Comment:
The reationale for using an arel instead of a scope here is so it could be shared between the subselect above that needs to have no from clause in order to add a boolean pseudocolumn, and the more normal-looking filtering we do in `exclude_incomplete_features_for`.

A note on the kind of finicky language - calling these things `excluding_incomplete_features` instead of `for_complete_features` or something: The rationale is that there are two paths here - either it's a feature gate, and it's complete, or it's not a feature gate at all (not a feature). Anyway, lemme know if you hate that.